Michael-Tu/cppGFS2.0
# [feat] - WriteRequest and ApplyMutation implementation for chunk server file service

This change includes how the primary chunk server handles write request, and ask replica chunk servers to apply mutation after primary is done writing.

Changes:
1. Implementation for WriteRequest and ApplyMutation
2. Test
3. Add definition for ApplyMutation grpc
4. Fix checksum fields in grpc to use bytes instead of string, because checksum sequence isn't valid UTF-8
5. Verify sent data checksum for SendDataRequests.
